src/Model.php

```diff
@@ -609,37 +625,50 @@ public function isDirty(string $field): bool
      *
      * @return Field[]
      */
-    public function getFields($filter = null): array
+    public function getFields($filters = null): array
     {
-        if ($filter === null) {
+        if ($filters === null) {
             return $this->fields;
```

Comment:
I do not like this mixed concept.

if null, then all
if for ex. `FIELD_FILTER_VISIBLE`, then not visible from all, but also with ONLY FIELDS filter

can we do something about this?

also, it seems that the new constants are not applied to other code

Reply:
I agree it is mixed and results are not intuitive
`FIELD_FILTER_VISIBLE`
`FIELD_FILTER_VISIBLE_ONLY_FIELDS`, etc?

Reply:
it will be not BC anyway, wdyt about additional `$filterOnlyFields` param? then probably no `FIELD_FILTER_ONLY_FIELDS` and `FIELD_FILTER_PERSIST`

Reply:
I played with that as well but we are in BC-break territory then.
if we set `$filterOnlyFields = true` by default (which is now the case for filter presets) it will not work correctly when `$filters = null`

Reply:
yes - but is this wanted?

Reply:
Logically not, historically - yes :)
Logically $filterOnlyFields would be false and all fields returned.
Then gradually filtered out

Reply:
Resolved